### ISSUE TITLE
runc {create,run}: --config is missing

### ISSUE THREAD

**Issue by kolyshkin** [opencontainers/runc#3308]
crun has added `--config` option as an ability to specify name other than `config.json` for the spec file in https://github.com/containers/crun/pull/310, and later changed its semantics a little, making it relative to the cwd (rather than `--bundle` argument), and allowing paths rather than just filenames (in https://github.com/containers/crun/pull/461).

I guess we should follow the suite.

Alas, in runc it's not possible to have `exec` or `update` in a container created with `--config`, because `runc exec` requires and reads `config.json`. It is different in crun, as its exec and update do not rely on `config.json`, using a smaller config written during creation instead (see https://github.com/opencontainers/runc/issues/3180).

Thus, if we want to fix it (and I think we do -- feature parity is important), we need to fix #3180 first.

**Comment by kolyshkin** [comment]
OK, I was wrong. `crun exec` loads config.json, but it does it from a copy saved to runtime dir (`/run/crun/$CTID/config.json`).

Maybe we should do the same thing.

**Comment by cyphar** [comment]
I'm a bit conflicted. The runtime-spec defines a runtime bundle and config.json:

> config.json: contains configuration data. This REQUIRED file MUST reside in the root of the bundle directory and MUST be named config.json. See config.json for more details.

On the other hand, I have talked to folks who use libcontainer directly purely so they don't need to create a dummy bundle directory to start containers. Being able to specify config.json would probably be a good thing -- but I don't think we should allow having a bundle that has a differently-named config.json.

Basically this would mean that if you specify --config, rootfs as well as all other paths must be absolute paths, to make it very clear there is no "bundle" directory in this configuration.

**Comment by kolyshkin** [comment]
I don't see a particularly good reason to have `--config` per se -- the reason I'm looking at it as I thought it would be a good idea to have crun tests run on runc. Alas, those tests use `--config` extensively. Perhaps I should instead look into fixing those tests so they would work without `--config`.

@giuseppe do you remember why `--config` is added, what issues can it solve, does it have any real users? I took a look at https://github.com/containers/crun/issues/308 and am not entirely convinced.

**Comment by giuseppe** [comment]
not sure if it is helpful, but as @cyphar described, I was playing with avoiding a dummy bundle directory, with `--config` you can do stuff like:

```sh
$ export ROOTFS=... # exploded rootfs
$ crun spec -b /tmp
$ crun run --config <(jq "(.process.args?) = [\"echo\", \"hello\"] | (.root.path?) =\"$ROOTFS\"" < /tmp/config.json) foo
hello
$ crun run --config <(jq "(.process.args?) = [\"echo\", \"world\"] | (.root.path?) =\"$ROOTFS\"" < /tmp/config.json) foo
world
```

Maybe `spec` needs a flag to write to stdout :-)
